alternative-routes/map.js

```diff
@@ -0,0 +1,201 @@
+  routeOptions.forEach((route, index) => route.addEventListener('change', () => changeRoute(routes, index)));
+};
+
+/**
+ * Draw a polyline on a map.
+ *
+ * @param routes {object} All routes recieved from the route query
+ * @param index {number} The index of the route that should be drawn
+ */
+const drawPolyline = (route, index, linecolor) => {
+  const geojson = {
+    type: 'FeatureCollection',
+    features: [
+      {
+        type: 'Feature',
+        geometry: {
+          type: 'LineString',
+          properties: {
+            description: route.data.duration + ' ' + route.data.charges,
+          },
+          coordinates: route.polyline,
+        },
+      },
+    ],
+  };
+
+  map.addSource(`${index}`, {
+    type: 'geojson',
+    lineMetrics: true,
+    data: geojson,
+  });
+
+  map.addLayer({
+    id: `${index}`,
+    type: 'line',
+    options: 'beforeLayer',
+    source: `${index}`,
+    layout: {
+      'line-join': 'round',
+      'line-cap': 'round',
+    },
+    paint: {
+      'line-color': linecolor,
+      'line-width': 5,
+    },
+  });
+};
+
+/**
+ * Show the charging station, origin and destination on the map.
+ *
+ * Last leg of the route is a destination point.
+ * All other legs are either charging stations or via points (if route has stops).
+ *
+ * @param legs {array} route legs
+ */
+const showLegs = legs => {
+  if (legs.length === 0) return;
+  if (map.getLayer('legs')) map.removeLayer('legs');
+  if (map.getSource('legs')) map.removeSource('legs');
+
+  let points = [];
+
+  // we want to show origin point on the map
+  // to do that we use the origin of the first leg
+  points.push({
+    type: 'Feature',
+    properties: {
+      icon: 'location_big',
+    },
+    geometry: legs[0].origin?.geometry,
+  });
+
+  legs.map((leg, index) => {
+    // add charging stations
+    if (index !== legs.length - 1) {
+      points.push({
+        type: 'Feature',
+        properties: {
+          description: `${getDurationString(leg.chargeTime)}`,
+          icon: 'unknown-turbo',
+        },
+        geometry: leg.destination?.geometry,
+      });
+    } else {
+      // add destination point (last leg)
+      points.push({
+        type: 'Feature',
+        properties: {
+          icon: 'arrival',
+        },
+        geometry: leg.destination?.geometry,
+      });
+    }
+  });
+
+  // draw route points on a map
+  map.addLayer({
+    id: 'legs',
+    type: 'symbol',
+    layout: {
+      'icon-image': '{icon}',
+      'icon-allow-overlap': true,
+      'icon-size': ['case', ['==', ['get', 'icon'], 'location-big'], ['literal', 0.7], ['literal', 0.8]],
+      'icon-offset': ['case', ['==', ['get', 'icon'], 'location_big'], ['literal', [0, 0]], ['literal', [0, -15]]],
+    },
+    source: {
+      type: 'geojson',
+      data: {
+        type: 'FeatureCollection',
+        features: points,
+      },
+    },
+  });
+};
+
+/**
+ * Highlight the route that was clicked.
```

Comment:
It says "highlight", but the name of the method says "change" If this method indeed highlights the route, I would then use it in the name of the method.